samples/JobsAgent/JobsAgent.cpp

```diff
@@ -0,0 +1,291 @@
+                AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT,
+                              "Error in Parsing. %s\n parse error code : %d, offset : %d",
+                              ResponseHelper::ToString(rc).c_str(),
+                              static_cast<int>(util::JsonParser::GetParseErrorCode(installed_packages_json_)),
+                              static_cast<unsigned int>(util::JsonParser::GetParseErrorOffset(installed_packages_json_)));
+                return rc;
+            }
+
+            rc = InitializeTLS();
+            if (ResponseCode::SUCCESS != rc) {
+                return rc;
+            }
+
+            ClientCoreState::ApplicationDisconnectCallbackPtr p_disconnect_handler =
+                std::bind(&JobsAgent::DisconnectCallback, this, std::placeholders::_1, std::placeholders::_2);
+
+            ClientCoreState::ApplicationReconnectCallbackPtr p_reconnect_handler =
+                std::bind(&JobsAgent::ReconnectCallback, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
+
+            ClientCoreState::ApplicationResubscribeCallbackPtr p_resubscribe_handler =
+                std::bind(&JobsAgent::ResubscribeCallback, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
+
+            p_iot_client_ = std::shared_ptr<MqttClient>(MqttClient::Create(p_network_connection_,
+                                                                           ConfigCommon::mqtt_command_timeout_,
+                                                                           p_disconnect_handler, nullptr,
+                                                                           p_reconnect_handler, nullptr,
+                                                                           p_resubscribe_handler, nullptr));
+            if (nullptr == p_iot_client_) {
+                return ResponseCode::FAILURE;
+            }
+
+            util::String client_id_tagged = ConfigCommon::base_client_id_;
+            client_id_tagged.append("_jobs_agent_");
+            client_id_tagged.append(std::to_string(rand()));
+            std::unique_ptr<Utf8String> client_id = Utf8String::Create(client_id_tagged);
+
+            rc = p_iot_client_->Connect(ConfigCommon::mqtt_command_timeout_, ConfigCommon::is_clean_session_,
+                                        mqtt::Version::MQTT_3_1_1, ConfigCommon::keep_alive_timeout_secs_,
+                                        std::move(client_id), nullptr, nullptr, nullptr);
+            if (ResponseCode::MQTT_CONNACK_CONNECTION_ACCEPTED != rc) {
+                return rc;
+            }
+
+            StartInstalledPackages();
+
+            p_jobs_ = Jobs::Create(p_iot_client_, mqtt::QoS::QOS1, ConfigCommon::thing_name_, client_id_tagged);
+
+            rc = Subscribe();
+            if (ResponseCode::SUCCESS != rc) {
+                AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT, "Subscribe failed. %s", ResponseHelper::ToString(rc).c_str());
```

Comment:
If you used the LOGSTREAM API,  ResponseHelper::ToString(rc) wouldn't be evaluated unless that log level is invoked.